packages/core/test/test-util/xsuaa-service-mocks.ts

```diff
@@ -1,20 +1,46 @@
 import nock from 'nock';
-import { basicHeader } from '../../src/connectivity/scp-cf';
+import { basicHeader, ServiceCredentials } from '../../src/connectivity/scp-cf';
```

Comment:
I put this comment here, but it is about the `xsuaa-service.ts` should we deprecate the `clientCredentialsGrant` method or also replace the implementation using the xssec?

Reply:
Currently this method will fail it the credential type is `x509`

Reply:
Probably yes, but I would do this on a separate PR. I would like to see E2E tests work an so on and then slowly (but soon) replace the rest. I think this can still be done as part of this ticket.